### DISCUSSION TITLE
deploy staging with directory protection

### DISCUSSION THREAD

**Discussion by ghost** [sourcebroker/deployer-extended-typo3#40] [Q&A]
I have been searching the internet for a while now but have not found a solution. I would like to protect the staging environment with htaccess and htpasswd. But then the deployment will no longer work. Is it even possible or do I need a different approach?

**Comment by kszymukowicz** [comment]
@eBornstein 

I suppose your deploy fails on task "cache:clear_php_http". 
https://github.com/sourcebroker/deployer-extended/blob/master/deployer/cache/task/cache_clear_php_http.php

By default the request to clear cache is done with curl and for curl you have possiblity to set any option you lie by setting
```
set('fetch_method_curl_options', '...curl options here');
```

So this way you can also set curl htaccess creds.

**Reply by ghost** [reply]
Thanks but the error was:
```
[staging] In Dotenv.php line 52:
[staging] [ParseError]
[staging] syntax error, unexpected 'static' (T_STATIC)
...
ERROR: Task deploy:vendors failed!
```

Solution for this error: 
`->set('fetch_method_curl_options', '--user user:pass') `
Afterward deployment fails on task "cache:clear_php_http" and solution for that: 
`->set('fetch_method_wget_options', '--user user --password pass')`

So with both lines it works great, thanks.
